File: DotNetAssignments/c#mini/ef-core/efcore/entities/Employee.cs
namespace efcore.entities;

public class Employee{
    //empno  ename  job  mgr  hiredate   sal  comm  deptno 
    public decimal Id { get; set; }
    public string Name { get; set; }
    public string Job { get; set; }
    public decimal mgr { get; set; }
    public DateTime HireDate { get; set; }
    public decimal Sal { get; set; }
    public decimal Comm { get; set; }
    // Navigation property
    public Department Department { get; set; }

    // One-to-Many relationship with Department
    public int DepartmentId { get; set; }


}

File: JavaAssignment/akash/JeanPaulAnomaly/src/main/java/app/data/Employee.java
package app.data;

import java.util.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "EMP")
public class Employee {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int empno;

    @Column
    private String ename;

    @Column
    private String job;
    
    @Column
    private int mgr;
    
    @Column
    private Date hiredate;

    @ManyToOne
    @JoinColumn(name = "deptno")
    private Department department;

    public int getEmpno() {
        return empno;
    }

    public void setEmpno(int empno) {
        this.empno = empno;
    }

    public String getEname() {
        return ename;
    }

    public void setEname(String ename) {
        this.ename = ename;
    }

    public String getJob() {
        return job;
    }

    public void setJob(String job) {
        this.job = job;
    }

    public int getMgr() {
        return mgr;
    }

    public void setMgr(int mgr) {
        this.mgr = mgr;
    }

    public Date getHiredate() {
        return hiredate;
    }

    public void setHiredate(Date hiredate) {
        this.hiredate = hiredate;
    }

    public Department getDepartment() {
        return department;
    }

    public void setDepartment(Department department) {
        this.department = department;
    }
    
    

}